### PR TITLE
Build CUDA 10.2 Support

### PR DESCRIPTION
**Pull request by jakirkham**
Partially addresses issue ( https://github.com/conda-forge/cupy-feedstock/issues/15 ).

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jakirkham** [comment]
I think we need to bump the global `cudnn` pin to `7.6.5`.

**Comment by leofang** [comment]
@jakirkham Thanks for working on this! I have a question: I've been following conda-forge/conda-forge-pinning-feedstock#360 but I still have no clue what's going on there. I don't understand: 1. what's a migration, and 2. what exactly prevents an auto-rendering from happening and requires a manual editing as you did here. Could you fill in the context for the rest of us when you have time please?

**Comment by jakirkham** [comment]
A migration is an automated rollout of PRs like this one following topological sort order (upstream packages first and then their dependencies). Unfortunately there appears to still be a bit of work to get the automation around migration to work correctly for CUDA upgrades. In this case it seemed easier to simply make the migration PRs manually, which is what we do. The migration is then followed by graduating that content into `conda-forge-pinning`.

Re-rendering is a different process which inspects `conda-forge-pinning` and the `conda_build_config.yaml` in the recipe here and use this to generate the contents of [`.ci_support`]( https://github.com/conda-forge/cupy-feedstock/tree/master/.ci_support ). The contain Python versions, CUDA versions, and other relevant constraints. It also ensures that CI's contain one build per file in `.ci_support`. Other content like the `README.md` is also generated in this process.

Running an automated re-render would still work here. It was just easier to do it locally when making these changes.

Does that help? 🙂

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+cudnn:
+  - 7.6.5
```
Comment (jakirkham) [inline recipe/conda_build_config.yaml]: Added here as a workaround until PR ( https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/375 ) lands.

**Comment by leofang** [comment]
> Does that help? 🙂

Definitely! 👍

File: recipe/conda_build_config.yaml (hunk 1 above)
Comment (leofang) [reply recipe/conda_build_config.yaml]: So it can be safely removed now?

Comment (jakirkham) [reply recipe/conda_build_config.yaml]: Yeah. We have to wait a little bit for `conda-forge-pinning` to be copied over to the CDN (~20mins) and then update the bot (~2mins). After that we can drop this.

Figure in the meantime we can wait for CI and see if this fixes things or if there is more work to do.

Comment (leofang) [reply recipe/conda_build_config.yaml]: Looks like CI is happy with 10.2.

Comment (jakirkham) [reply recipe/conda_build_config.yaml]: Have reverted now 🙂

**Comment by leofang** [comment]
@jakirkham shouldn't we bump the build number?

**Comment by jakirkham** [comment]
> @jakirkham shouldn't we bump the build number?

Normally adding a new matrix element would not have a build number bump. However as we are bumping the version of `cudnn` we do need one. Thanks for catching that. Have pushed a build number bump here and to the v6 PR. Please let me know if there is anything else.

**Comment by jakirkham** [comment]
Thanks for the review and help @leofang! 😄

**Pull request merged**